OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -0,0 +1,816 @@
+    // debug marker, as they may be spanned across different submissions.
+    if (queue_submission_optional.has_value()) {
+      queue_submission_optional->meta_information.post_submission_cpu_timestamp =
+          MonotonicTimestampNs();
+    }
+
+    std::vector<uint32_t> marker_slots_to_reset;
+    VkDevice device = VK_NULL_HANDLE;
+    for (uint32_t submit_index = 0; submit_index < submit_count; ++submit_index) {
+      VkSubmitInfo submit_info = submits[submit_index];
+      for (uint32_t command_buffer_index = 0; command_buffer_index < submit_info.commandBufferCount;
+           ++command_buffer_index) {
+        VkCommandBuffer command_buffer = submit_info.pCommandBuffers[command_buffer_index];
+        if (device == VK_NULL_HANDLE) {
+          CHECK(command_buffer_to_device_.contains(command_buffer));
+          device = command_buffer_to_device_.at(command_buffer);
+        }
+        CHECK(command_buffer_to_state_.contains(command_buffer));
+        CommandBufferState& state = command_buffer_to_state_.at(command_buffer);
+
+        for (const Marker& marker : state.markers) {
+          std::optional<SubmittedMarker> submitted_marker = std::nullopt;
+          if (marker.slot_index.has_value() && queue_submission_optional.has_value()) {
+            submitted_marker = {.meta_information = queue_submission_optional->meta_information,
+                                .slot_index = marker.slot_index.value()};
+          }
+
+          switch (marker.type) {
+            case MarkerType::kDebugMarkerBegin: {
+              if (queue_submission_optional.has_value() && marker.slot_index.has_value()) {
+                ++queue_submission_optional->num_begin_markers;
+              }
+              CHECK(marker.label_name.has_value());
+              CHECK(marker.color.has_value());
+              MarkerState marker_state{.label_name = marker.label_name.value(),
+                                       .color = marker.color.value(),
+                                       .begin_info = submitted_marker,
+                                       .depth = markers.marker_stack.size(),
+                                       .depth_exceeds_maximum = marker.cut_off};
+              markers.marker_stack.push(std::move(marker_state));
+              break;
+            }
+
+            case MarkerType::kDebugMarkerEnd: {
+              MarkerState marker_state = markers.marker_stack.top();
+              markers.marker_stack.pop();
+
+              // If there is a begin marker slot from a previous submission, this is our chance to
+              // reset the slot.
+              if (marker_state.begin_info.has_value() && !queue_submission_optional.has_value()) {
+                marker_slots_to_reset.push_back(marker_state.begin_info.value().slot_index);
+              }
+
+              // If the begin marker was cut off, but the end marker was not (as it is in a
```

Comment:
What does "cut off" here mean? In time, or due to depth? (Exceeded maximum depth?)

Reply:
Change the wording a bit.